epoch.go

```diff
@@ -2232,31 +2311,40 @@ func (e *Epoch) triggerEmptyBlockNotarization(round uint64) {
 
 	e.Comm.Broadcast(&Message{EmptyVoteMessage: &signedEV})
 
-	e.addEmptyVoteRebroadcastTimeout(&signedEV)
+	e.addEmptyVoteRebroadcastTimeout()
 
 	if err := e.maybeAssembleEmptyNotarization(); err != nil {
 		e.Logger.Error("Failed assembling empty notarization", zap.Error(err))
 		e.haltedError = err
 	}
 }
 
-func (e *Epoch) addEmptyVoteRebroadcastTimeout(vote *EmptyVote) {
-	task := &TimeoutTask{
-		NodeID:   e.ID,
-		TaskID:   EmptyVoteTimeoutID,
-		Deadline: e.timeoutHandler.GetTime().Add(e.EpochConfig.MaxRebroadcastWait),
-		Task: func() {
-			e.Logger.Debug("Rebroadcasting empty vote because round has not advanced", zap.Uint64("round", vote.Vote.Round))
-			e.Comm.Broadcast(&Message{EmptyVoteMessage: vote})
-			e.addEmptyVoteRebroadcastTimeout(vote)
-		},
+func (e *Epoch) emptyVoteTimeoutTaskRunner(_ []string) {
+	e.lock.Lock()
+	roundVotes, ok := e.emptyVotes[e.round]
```

Comment:
we need to lock for the entirety of this function, not just here. 

What if `roundVotes.votes` is accessed concurrently?